src/tokenizedSubaccounts/BaseTSA.sol

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.18;
+
+import {StandardManager} from "v2-core/src/risk-managers/StandardManager.sol";
+import {PMRM, IPMRM} from "v2-core/src/risk-managers/PMRM.sol";
+import {IBaseManager} from "v2-core/src/interfaces/IBaseManager.sol";
+import {IWrappedERC20Asset} from "v2-core/src/interfaces/IWrappedERC20Asset.sol";
+import {ILiquidatableManager} from "v2-core/src/interfaces/ILiquidatableManager.sol";
+import {ISubAccounts} from "v2-core/src/interfaces/ISubAccounts.sol";
+import {IMatching} from "../interfaces/IMatching.sol";
+import {IAsset} from "v2-core/src/interfaces/IAsset.sol";
+import {DutchAuction} from "v2-core/src/liquidation/DutchAuction.sol";
+import {ERC20} from "openzeppelin/token/ERC20/ERC20.sol";
+import {IERC20Metadata} from "openzeppelin/token/ERC20/extensions/IERC20Metadata.sol";
+import {BaseModule} from "../modules/BaseModule.sol";
+import {Ownable2Step} from "openzeppelin/access/Ownable2Step.sol";
+import {ConvertDecimals} from "lyra-utils/decimals/ConvertDecimals.sol";
+
+// TODO: handle valuation when there is an insolvency
+abstract contract BaseTSA is ERC20, Ownable2Step {
```

Comment:
can we add a read function, similar to above that takes in a list of addresses and returns list of token balances?